### PR TITLE
CSS for autohiding address/bookmark bar in minimalist

### PR DESCRIPTION
**Pull request by mbartlet**
This causes only the tab bar to be visible when interacting with the webpage DOM. If the user focuses or mouse-hovers the tabs, it drops down the address bar and the bookmarks below it. The mouse can be placed just slightly beneath the address bar but on the bookmark bar to vanish the address bar and select bookmarks as normal. This attempts to leave as much vertical space in the browser window as reasonably possible for the actual web page content, with the opinionation being that tabs are the only essential UI component that should be visible at all times.

### CODE REVIEW

**Comment by tiosgz** [comment]
Cool, I didn’t think anyone would contribute to this repo… but sometimes even unexpected things happen. Btw., I have to disagree with you:
> tabs are the only essential UI component that should be visible at all times

I have even the tabs hidden.

***

Try moving the bars around
* if either the tabs or the address bar aren’t on top, the bookmarks bar covers the page
* placing the address bar at the bottom isn’t respected
* maybe something else I left unnoticed

If you cannot make it work for the other configurations, just use only the working part (e.g. only auto-hide the bookmarks bar) & rename the file to include `top` (if I were to name it, I’d use `autohide_top_address_bookmark_bar.css`)

**Comment by mbartlet** [comment]
I knew someone would disagree with my tab opinion one day :)

I've incorporated your feedback and separated out the functionality, as well as provided better naming. I've left the original combined version as well because the autohide-bookmark-bar css needs different values to cooperate with the autohide-address-bar. This should make it a bit more modular.

Curiously the focus events don't seem to fire when the tabs are anywhere but the top for some reason (I added the appropriate left/right/bottom classes to the selector clause), which resulted in both bars being permanently hidden. If I'm able to get it working I will make an additional commit

**Comment by tiosgz** [comment]
Oh, sorry, I’ve expressed myself wrong. When I said
> If you cannot make it work for the other configurations, just use only the working part

I meant *try to make it work also with the other positioning, but if it isn’t easily possible, just disable the mod for them so that it UI works*, not *separate the parts*. I think auto-hiding both of them is OK, if someone wanted to hide only one, they can change it themself. Please delete the separate files.

Sorry for adding you this unnecessary work you’ve done.

**Comment by mbartlet** [comment]
I have removed the extra files and I will try to make the existing mod only modify the UI if the addres-bar and bookmark-bar are at the top

**Comment by mbartlet** [comment]
I tested with bookmark and address bar on the bottom, this new change only causes the auto-hiding if both are on top. It also appears to work well whether tabs are visible or not

File: css/minimalist/autohide_top_address_bookmark_bar_visible_tabs_top.css [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+#browser.tabs-top.address-top.bookmark-bar-top .bookmark-bar {
+    position: absolute !important;
+    width: 100%;
+    transition: transform 0.2s !important;
+    z-index: 1;
+}
+
+/* tabs-top address-top bookmark-bar-top */
```
Comment (tiosgz) [inline css/minimalist/autohide_top_address_bookmark_bar_visible_tabs_top.css:58]: Do you still need this comment?

**Review by tiosgz — CHANGES_REQUESTED** [review]
Probably the last change before I merge it—
* it messes up theming (try to keep the original colours if you don’t want to add another zillion lines)
* some of the selectors are still applied even if nothing is at the top

I still haven’t read the code continuously, but I’ll do that when I clean up the repo (hopefully).

**Comment by mbartlet** [comment]
> Probably the last change before I merge it—
> 
> * it messes up theming (try to keep the original colours if you don’t want to add another zillion lines)
> * some of the selectors are still applied even if nothing is at the top
> 
> I still haven’t read the code continuously, but I’ll do that when I clean up the repo (hopefully).

Can you elaborate on how it messes with the theming, or how I can replicate? I have mine set to change colors with the page and the colors seem to behave the same as vanilla--is that not what you mean?

And by "nothing at the top" I assume you mean tab-bar, address-bar, and bookmark-bar all set to hidden and/or not-top? I will test further

**Comment by tiosgz** [comment]
> Can you elaborate on how it messes with the theming, or how I can replicate? I have mine set to change colors with the page and the colors seem to behave the same as vanilla--is that not what you mean?

Hmm… it seems it’s just this colouring the address bar with accent when it’s at the bottom—so it’ll be okay when you fix the other issue.
```css
#browser:not(.color-behind-tabs-off) .UrlBar {
￼    background: inherit;
￼}
```
> And by "nothing at the top" I assume you mean tab-bar, address-bar, and bookmark-bar all set to hidden and/or not-top? I will test further

Yep. Basically all you have to do is check that every selector has the three classes (`.address-top`, `.bookmark-bar-top`, `.tabs-top`) and if it breaks the mod in any way when you add them (maybe you can completely omit `.tabs-top`; the address bar can still auto-hide even when the bookmark bar isn’t at the top &c).

**Comment by mbartlet** [comment]
I've removed redundant comments, and tested that all styling is now applied _only_ when all 3 bars are not hidden and configured to be placed at the top of the window

**Comment by mbartlet** [comment]
For what it's worth just leaving this here, it's the command flags to enable vivaldi debug mode which enables console-UI level inspector (right click on the tabs/bookmark bar itself and select "Inspect') that greatly simplified this process
`vivaldi-stable --flag-switches-begin --debug-packed-apps --silent-debugger-extension-api --flag-switches-end`

File: css/minimalist/autohide_top_address_bookmark_bar_visible_tabs_top.css (hunk 1 above)
Comment (mbartlet) [reply css/minimalist/autohide_top_address_bookmark_bar_visible_tabs_top.css]: comments removed, none of them were informative

**Review by tiosgz — APPROVED** [review]
Now it seems a little bit too restrictive, but I’ll merge it & do the changes myself.

**Pull request merged**